torch_points3d/models/base_model.py

```diff
@@ -1,86 +1,16 @@
-        # compute_warmup needs the datamodule to be available when `self.num_training_steps`
-        # is called that is why this is done here and not in the __init__
-        self.scheduler_cfg.num_training_steps, self.scheduler_cfg.num_warmup_steps = self.compute_warmup(
-            num_training_steps=self.scheduler_cfg.num_training_steps,
-            num_warmup_steps=self.scheduler_cfg.num_warmup_steps,
-        )
-        rank_zero_info(f"Inferring number of training steps, set to {self.scheduler_cfg.num_training_steps}")
-        rank_zero_info(f"Inferring number of warmup steps from ratio, set to {self.scheduler_cfg.num_warmup_steps}")
-        self.scheduler = self.instantiator.scheduler(self.scheduler_cfg, self.optimizer)
-
-        return {
-            "optimizer": self.optimizer,
-            "lr_scheduler": {"scheduler": self.scheduler, "interval": "step", "frequency": 1},
-        }
-
-    @property
-    def num_training_steps(self) -> int:
-        """Total training steps inferred from datamodule and devices."""
-        if isinstance(self.trainer.limit_train_batches, int) and self.trainer.limit_train_batches != 0:
-            dataset_size = self.trainer.limit_train_batches
-        elif isinstance(self.trainer.limit_train_batches, float):
-            # limit_train_batches is a percentage of batches
-            dataset_size = len(self.trainer.datamodule.train_dataloader())
-            dataset_size = int(dataset_size * self.trainer.limit_train_batches)
-        else:
-            dataset_size = len(self.trainer.datamodule.train_dataloader())
 
-        num_devices = max(1, self.trainer.num_gpus, self.trainer.num_processes)
-        if self.trainer.tpu_cores:
-            num_devices = max(num_devices, self.trainer.tpu_cores)
-
-        effective_batch_size = self.trainer.accumulate_grad_batches * num_devices
-        max_estimated_steps = (dataset_size // effective_batch_size) * self.trainer.max_epochs
-
-        if self.trainer.max_steps and self.trainer.max_steps < max_estimated_steps:
-            return self.trainer.max_steps
-        return max_estimated_steps
-
-    def compute_warmup(self, num_training_steps: int, num_warmup_steps: Union[int, float]) -> Tuple[int, int]:
-        if num_training_steps < 0:
-            # less than 0 specifies to infer number of training steps
-            num_training_steps = self.num_training_steps
-        if isinstance(num_warmup_steps, float):
-            # Convert float values to percentage of training steps to use as warmup
-            num_warmup_steps *= num_training_steps
-        return num_training_steps, num_warmup_steps
+        self.instantiator = instantiator
 
-    def setup(self, stage: str):
-        self.configure_metrics(stage)
+    def set_input(self, data):
+        raise (NotImplementedError("set_input needs to be defined!"))
 
-    def configure_metrics(self, stage: str) -> Optional[Any]:
```

Comment:
maybe we should also have a function that return all the losses
For some models such as KPConv, there are specific [regularization losses.](https://github.com/nicolas-chaulet/torch-points3d/blob/5378a2ae7ef612433acedaa497073971ec66ff09/torch_points3d/models/segmentation/kpconv.py#L142)
. The forward will return the total loss.
```python
def get_losses(self) -> Dict[str, torch.Tensor]:
"""
"""
```
When we have multiple losses we can track them.

Reply:
Good idea!